store/store.go

```diff
@@ -634,8 +648,19 @@ func (s *Store) retryRequest(ctx context.Context, client *http.Client, reqOption
 		if shouldRetryHttpResponse(attempt, resp) {
 			resp.Body.Close()
 			continue
+		} else {
+			err = decode(resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusCreated, resp.Body)
+			if err != nil {
+				resp.Body.Close()
```

Comment:
I think we want this above, right after decode, so that we always close the body. We're now consuming it here, so it makes sense to close here as well. This means we can drop the Body.Close() everywhere we use this function.